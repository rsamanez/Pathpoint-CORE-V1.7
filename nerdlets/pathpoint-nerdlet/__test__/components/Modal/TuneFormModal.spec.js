import React from 'react';
import { create } from 'react-test-renderer';
import {
  BodyTuneFormModal,
  HeaderTuneFormModal
} from '../../../components/Modal/TuneFormModal';

describe('TuneFormModal component', () => {
  test('Render body with default data', () => {
    const body = create(
      <BodyTuneFormModal
        stageNameSelected={{
          selectedCase: {
            value: 0
          },
          datos: [
            {
              label: 'Full Open Query',
              query_body: 'SELECT FILTER(count(*) FROM Log',
              query_footer: 'SINCE 5 MINUTES AGO',
              query_start: '',
              type: 20,
              value: 0
            }
          ]
        }}
        handleSaveUpdateTune={jest.fn()}
        handleSubmitTune={jest.fn()}
        handleOnChange={jest.fn()}
      />
    );
    expect(body.toJSON()).toMatchSnapshot();
  });

  test('Render header with default data', () => {
    const header = create(
      <HeaderTuneFormModal
        stageNameSelected={{
          touchpoint: {
            value: 'touchpoint one'
          },
          selectedCase: {
            value: 0
          },
          datos: [
            {
              label: 'Full Open Query',
              query_body: 'SELECT FILTER(count(*) FROM Log',
              query_footer: 'SINCE 5 MINUTES AGO',
              query_start: '',
              type: 20,
              value: 0
            }
          ]
        }}
      />
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
