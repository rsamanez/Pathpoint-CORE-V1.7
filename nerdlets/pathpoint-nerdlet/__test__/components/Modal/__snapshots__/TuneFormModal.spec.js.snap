// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TuneFormModal component Render body with default data 1`] = `
<div
  style={
    Object {
      "width": "auto",
    }
  }
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "end",
          "marginBottom": "1px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "width": "47%",
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <label
            className="headerSubtitleTune"
          >
            Configured
          </label>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "center",
              "width": "40%",
            }
          }
        >
          <label
            className="headerSubtitleTune"
          >
            Last 5 min
          </label>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-end",
        }
      }
    >
      <button
        className="btn btn-default"
        color="primary"
        disabled={false}
        style={
          Object {
            "background": "#0178bf",
            "color": "white",
          }
        }
        type="submit"
        variant="contained"
      >
        Save / Update
      </button>
    </div>
  </form>
</div>
`;

exports[`TuneFormModal component Render header with default data 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    className="titleModal"
  >
    touchpoint one
  </div>
</div>
`;
